tests/python/pytorch/sparse/test_sparse_matrix.py

```diff
@@ -450,6 +450,52 @@ def test_has_duplicate():
     assert csc_A.has_duplicate()
 
 
+def test_select():
+    ctx = F.ctx()
+
+    row = torch.tensor([0, 1, 1, 2, 3, 4]).to(ctx)
```

Comment:
Hardcode test cases only cover very limited scenarios. You can try to parameterize the test cases. https://github.com/dmlc/dgl/blob/ec8225df4ab164cc37522fd1a903ba00d1aed26c/tests/python/pytorch/sparse/test_sparse_matrix.py#L88 is a good example. And you also need to use some other tool to produce the correct result for each case. You can try to convert your results to pytorch dense tensor and verify the correctness by Pytorch dense index_select.

Reply:
ok, I have extended tests by generating correct results automatically. please review.